pyproject.toml

```diff
@@ -72,11 +72,21 @@ ignore = [
 ]
 
 [tool.ruff.lint.per-file-ignores]
-# Ignore `E402` (import violations) in all examples
-"examples/**" = ["E402"]
+"examples/**" = [
+    # Ignore `E402` (import violations) in all examples
+    "E402", 
+    # Ignore missing required imports
+    "I002"
+    ]
+"docs/**" = [
+    # Ignore missing required imports
+    "I002"
+    ]
 
 [tool.ruff.lint.isort]
 known-third-party = ["datasets"]
+required-imports = ["from __future__ import annotations"]
```

Comment:
Perfect, was looking for this. Otherwise I can't move forward with the typing changes without breaking backwards compatibility.